### PR TITLE
Logging for matcher module

### PR DESCRIPTION
**Pull request by gsn9**
This might be a longer pr as I did all of the matcher module's files in one go. I also added logging in several places where we are raising a value error. I can remove them if it's unneeded but I was thinking it would be good to log anytime a value error occurs.

### CODE REVIEW

File: autocnet/matcher/ciratefi.py [hunk 1]
```diff
@@ -75,17 +76,19 @@ def cifi(template, search_image, thresh=90, use_percentile=True,
 
     radii = np.asarray(radii)
     if not radii.size or not np.any(radii):
+        log.error('Input radii list is empty')
         raise ValueError('Input radii list is empty')
```
Comment (gsn9) [inline autocnet/matcher/ciratefi.py:81]: these are the log's I mentioned before

File: autocnet/matcher/tests/test_ciratefi.py [hunk 2]
```diff
@@ -51,10 +52,16 @@ def offset_template(img, img_coord):
...
+def test_cifi_radii_too_large(template, search, caplog):
+    # check all logs
+    ciratefi.cifi(template, search, 1.0, radii=[100], use_percentile=False)
+    num_pattern = '\d+'
+    captured_log = caplog.records[0].getMessage()
+    match = re.findall(num_pattern, captured_log)
+    assert (f'Max Radii is larger than original template, '\
+        f'this may produce sub-par results.Max radii: {match[0]} max template dimension: {match[1]}'\
+             == caplog.records[0].getMessage())
+
```
Comment (gsn9) [inline autocnet/matcher/tests/test_ciratefi.py:64]: I kinda changed the way this test is testing for the right log/warning now. With using logging I didn't find a simple way to pull the template dimensions so rather then going down that rabbit hole I decided to use a regular expression. Basically I'm just searching the captured log for the dimensions and then just checking to make sure that the expected matches the results.

File: autocnet/matcher/tests/test_outlier_detector.py [hunk 3]
```diff
@@ -65,11 +64,11 @@ def test_suppress(self):
         self.assertEqual(mask.sum(), 34)
 
     def test_suppress_non_optimal(self):
-        with warnings.catch_warnings(record=True) as w:
+        # with warnings.catch_warnings(record=True) as w:
+        with self.assertLogs() as captured:
             mask, k = cpu_outlier_detector.spatial_suppression(self.df, self.domain, k=30, xkey='lon', ykey='lat')
             self.assertEqual(k, 59)
-            self.assertTrue(issubclass(w[0].category, UserWarning))
```
Comment (gsn9) [inline autocnet/matcher/tests/test_outlier_detector.py:71]: I removed this line as I removed the warnings import, I think this should be okay?

**Review by Kelvinrr — APPROVED** [review]
Looks reasonable to me assuming CI failures are expected.

**Pull request merged**